pkg/api/defaults.go

```diff
@@ -233,9 +233,19 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpgrade, isScale bool) {
 		if "" == a.OrchestratorProfile.KubernetesConfig.EtcdDiskSizeGB {
 			switch {
 			case a.TotalNodes() > 20:
-				a.OrchestratorProfile.KubernetesConfig.EtcdDiskSizeGB = DefaultEtcdDiskSizeGT20Nodes
+				if a.IsAzureStackCloud() {
+					// Currently on Azure Stack max size of managed disk size is 1023GB.
```

Comment:
@rjaini do we know how much disk is required for 20 + nodes?  The cluster will be not usable if the disc is out of space.
Another alternative is to limit the total count of agent nodes.
And we should raise the request to Disk RP on Azure Stack to support larger size of disk.

Reply:
I am not sure how much disk space requirements. But from https://github.com/Azure/aks-engine/issues/923 thread it looks like the 1024 size itself is very high.
I will check with Disk RP team to understand the reason of 1023 limit.

@mboersma : Do you know why we have such large disk size for higher number of nodes in AKS Engine ?.

Reply:
Matt is out on vacation but see https://github.com/Azure/acs-engine/pull/2435 for context.